core/src/main/scala/pickling/Macros.scala

```diff
@@ -334,9 +342,9 @@ trait OpenSumUnpicklerMacro extends Macro with UnpicklerMacros with FastTypeTagM
     def dispatchLogic = {
       val compileTimeDispatch = createCompileTimeDispatch(tpe)
       val refDispatch         = createRefDispatch()
+      // TODO - for now we're using "currentMirror" macro.  That may not be ok.
```

Comment:
It's OK. The design of `reader.mirror` was only half-baked, anyway, so I think it's better to remove it altogether for now.